### PR TITLE
Noble integer

### PR DESCRIPTION
**Pull request by rohanbabbar04**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Added Noble Integer in Mathematics section in C,C++ and Python.

### Checklist

- [ ✅] I've read the contribution guidelines.
- [✅ ] I've checked the issue list before deciding what to submit.
- [ ✅] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Added #2371

### CODE REVIEW

**Comment by rohanbabbar04** [comment]
Do review this PR @bhav09  @HarshCasper

**Comment by rohanbabbar04** [comment]
Could you please review the PR @ankitaggarwal23

**Review by ankitaggarwal23 — APPROVED** [review]

File: C-Plus-Plus/math/Noble_Integer.cpp [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	int num, i, x, arr[10000];
+	cout << "Enter the number of elements:";
+	cin >> num;
+	cout << "Enter the elements:";
+	for (int i = 0; i < num; i++)
+	{
+		cin >> x;
+		arr[i] = x;
+	}
+
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/math/Noble_Integer.cpp]: Remove the extra line.

Comment (ankitaggarwal23) [inline C-Plus-Plus/math/Noble_Integer.cpp]: Remove the extra line.

File: Python/math/Noble_Integer.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    for i in range(0,num-1):
+        if(lst[i]==lst[i+1]):
+            continue;
+        if(lst[i]==num-i-1):
+            x=1;
+            return(num-i-1);
+            break;
+    if(x==0):
+        return(-1);
+if __name__ == "__main__" :
```
Comment (ankitaggarwal23) [inline Python/math/Noble_Integer.py]: Give a new line after `return(-1);`

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
- Add Time and Space Complexity at the end of the Code

**Review by ankitaggarwal23 — APPROVED** [review]

**Comment by rohanbabbar04** [comment]
I have made all the changes do review @ankitaggarwal23 @HarshCasper

**Comment by rohanbabbar04** [comment]
Do approve the changes and merge @HarshCasper

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**